src/hexdoc/patchouli/page/pages.py

```diff
@@ -78,6 +79,19 @@ class Multiblock(HexdocModel):
     symmetrical: bool = False
     offset: tuple[int, int, int] | None = None
 
+    @property
+    def bill_of_materials(self) -> list[tuple[ItemWithTexture | TagWithTexture, int]]:
+        bom = defaultdict[str, int](int)
+        for layer in self.pattern:
+            for row in layer:
+                for item in row:
+                    if item in self.mapping:
+                        bom[item] += 1
+        bom_list = list[tuple[ItemWithTexture | TagWithTexture, int]]()
+        for item_key, count in bom.items():
+            bom_list.append((self.mapping[item_key], count))
+        return bom_list
```

Comment:
maybe something like this? (untested)
```suggestion

        return sorted(
            (self.mapping[item], count) for item, count in bom.items(),
            key=lambda v: (v[1], v[0].name),
        )
```